homeassistant/components/media_player/mpd.py

```diff
@@ -181,13 +183,34 @@ def supported_media_commands(self):
         """Flag of media commands that are supported."""
         return SUPPORT_MPD
 
+    @property
+    def source(self):
+        """Name of the current input source."""
+        return self.currentplaylist
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self.playlists
+
+    def select_source(self, source):
+        """Choose a different available playlist and play it."""
+        self.play_media(MEDIA_TYPE_PLAYLIST, source)
+
     def turn_off(self):
         """Service to send the MPD the command to stop playing."""
         self.client.stop()
 
     def turn_on(self):
         """Service to send the MPD the command to start playing."""
         self.client.play()
+        self._update_playlists()
```

Comment:
Great call! I copied it to there so playlists update while the component is turned on. Getting playlists from mopidy has a bit of a lag on my Pi so I throttled it quite a bit. I also kept it in the turn_on method so if people add a new playlist and want it immediately they can just power-cycle the component.